core/game/board.go

```diff
@@ -0,0 +1,161 @@
+package game
+
+import (
+	"container/list"
+	"fmt"
+	"strings"
+
+	"github.com/otrego/clamshell/core/color"
+	"github.com/otrego/clamshell/core/point"
+)
+
+// Board Contains the board, capturesStones, and ko
+// capturedStones only contains the stones captured
+// from the most recent AddStone call. ko contains
+// a point that is illegal to recapture due to Ko.
+type Board struct {
+	board          [][]color.Color
+	capturedStones []*point.Point
+	ko             *point.Point
+}
+
+// NewBoard creates a new size x size board.
+func NewBoard(size int) *Board {
+	board := Board{
+		make([][]color.Color, size),
+		nil,
+		nil,
+	}
+
+	for i := 0; i < size; i++ {
+		board.board[i] = make([]color.Color, size)
+	}
+	return &board
+}
+
+// AddStone adds a stone to the board.
+// returns the captured stones, or err
+// if any go (baduk) rules were broken
+func (b *Board) AddStone(m *Move) ([]*point.Point, error) {
+	pt := m.Point()
+	var (
+		x, y int          = int(pt.X()), int(pt.Y())
+		ko   *point.Point = b.ko
+	)
+	b.ko = nil
+	b.capturedStones = nil
+
+	if x >= len(b.board[0]) || y >= len(b.board) ||
+		x < 0 || y < 0 {
+		return nil, fmt.Errorf("move %v out of bounds for %dx%d board",
+			pt, len(b.board[0]), len(b.board))
+	}
+	if b.board[y][x] != color.Empty {
+		return nil, fmt.Errorf("move %v already occupied", pt)
+	}
+
+	b.board[y][x] = m.Color()
+	b.FindCapturedGroups(m)
```

Comment:
this should return the captured groups rather than dumping into a field in the struct.